src/components/Popup/OfficePopup.vue

```diff
@@ -0,0 +1,100 @@
+<template>
+    <div
+            ref="self"
+            :class="classNames.root"
+            @focus="focused = true"
+            @blur="onSelfBlur"
+    >
+        <slot/>
+    </div>
+</template>
+
+<script lang="ts">
+    import {Component, Vue, Prop} from "vue-property-decorator";
+    import {mergeStyleSets} from "@uifabric/merge-styles";
+    import {getDocument} from "@/utility/dom";
+
+    @Component
+    export default class OfficePopup extends Vue {
+        @Prop({type: Boolean, default: false}) private needsVerticalScrollBar!: boolean;
+        @Prop({type: Boolean, default: true}) private shouldRestoreFocus!: boolean;
+        private originalFocusedElement?: HTMLElement;
+
+        private focused: boolean = false;
+        private needsVerticalScrollBarCalc?:boolean;
+
+        get classNames() {
+            return mergeStyleSets({
+                root: [
+                    this.needsVerticalScrollBarCalc && {
+                        overflowY: "scroll"
+                    },
+                    !this.needsVerticalScrollBarCalc && {
```

Comment:
I think it would be enough to just have 
```
this.needsVerticalScrollBarCalc && {
    overflowY: "scroll"
},
```

I think you don't need the negation to set `overflowY` to `undefined`

Reply:
from my pov we have to, what if the oberflowY is set on the style of the component.
this would be applied to the root element and would end up having overflow something, instead of nothing.
may be tri state would be so wrong or?
true false handling styling intern.
undefined take whatever

Reply:
Yep, but if "overflowY" is set by the user, then it is set directly on the `div`, so you cannot overwrite it by using the css class

Reply:
Also the styles set by the user shall always have a higher priority and overwrite the component styles

Reply:
ah is the mergestyles only appling on top of provided styles

Reply:
or is it inserting it over user set styles

Reply:
`mergeStyles` only creates CSS classes based on the styles you put in. It returns these css class names which you bind to in the template.

Reply:
So it has the normal specificity of a CSS class, which will get overwritten by inline styles